analyses/fusion_filtering/04-project-specific-filtering.Rmd

```diff
@@ -0,0 +1,269 @@
+params:
+  histology:
+    label: "Clinical file" 
+    value: data/pbta-histologies.tsv
+    input: file
+  group:
+    label: "Grouping variable"
+    value: broad_histology
+    input: string
+  dataStranded:
+    label: "Input filtered fusion dataframe"
+    value: scratch/standardFusionStrandedExp_QC_expression_GTExComparison_annotated.RDS
+    input: file
+  dataPolya:
+    label: "Input filtered fusion dataframe"
+    value: scratch/standardFusionPolyaExp_QC_expression_GTExComparison_annotated.RDS
+    input: file    
+  numCaller:
+    label: "Least Number of callers to have called fusion"
+    value: 2
+    input: integer
+  numSample:
+    label: "Least Number of samples to have fusion per group"
+    value: 2
+    input: integer
+  numGroup:
+    label: "Max number of groups found in"
+    value: 1
+    input: integer
+  limitMultiFused:
+    label: "Max number of times gene can be fused per sample"
+    value: 5
+    input: integer     
+
+    
+---
+
+K S Gaonkar
+
+Filtered Fusions:
+1. In-frame fusions is called in atleast 2 samples per histology OR
+2. In-frame fusions is called in atleast 2 callers 
+AND
+Filtered-fusions found in more than 1 histology OR
+Filtered-fusion doesn't have multi-fused gene (more than 5 times in sample)
+
+Putative Driver:
+Filtering for general cancer specific genes
+Fusions with genes in either onco
+
+This notebook assumes you are in OpenPBTA-analysis project folder structure.
+
+
+```{r}
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+setwd(root_dir)
+#load required packages
+suppressPackageStartupMessages(library("readr"))
+suppressPackageStartupMessages(library("tidyverse"))
+suppressPackageStartupMessages(library("reshape2"))
+suppressPackageStartupMessages(library("qdapRegex"))
+
+#read filtFusion files
+strandedQCGeneFiltered_filtFusion<-readRDS(params$dataStranded)
+polyaQCGeneFiltered_filtFusion<-readRDS(params$dataPolya)
+
+QCGeneFiltered_filtFusion<-rbind(strandedQCGeneFiltered_filtFusion,polyaQCGeneFiltered_filtFusion)
+
+write.table(QCGeneFiltered_filtFusion,"analyses/fusion_filtering/results/FilteredFusion.tsv",sep="\t",quote=FALSE,row.names = FALSE)
+
+# subset for recurrent fusion detection and multifused genes QC
+fusion_calls<-unique(QCGeneFiltered_filtFusion)
+# remove distance from intergenic fusions
+fusion_calls$FusionName<-unlist(lapply(fusion_calls$FusionName,function(x) rm_between(x, "(", ")", extract = F)))
+
+
+# get grouping column id
+group<-params$group
+
+# get histology file
+clinical<-read.delim(params$histology,stringsAsFactors = FALSE)
+
+# Least number of callers
+numCaller<-params$numCaller
+
+# Least number of samples per group
+numSample<-params$numSample
+
+# Max number of groups 
+numGroup<-params$numGroup
+
+# Max number of times gene can be fused per sample
+limitMultiFused<-params$limitMultiFused
+
+print("Raw calls from STARfusion and Arriba for PBTA")
+table(fusion_calls$Caller)
+```
+
+
+```{r}
+
+# aggregate caller 
```

Comment:
This particular step had a problem in the earlier PR #166 (see https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/166/files#r340102734) where you used `group_by(FusionName, Sample )` instead of `group_by(FusionName, Sample ,Fusion_Type)`.

Generally, in earlier steps I think you use `group_by(FusionName, Sample)`, e.g.,

https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/54a8a1ae9d91881aed0360df1ac47df8d940c139/analyses/fusion_filtering/02-fusion-filtering.R#L221

Is not including `Fusion_Type` going to be problematic in those steps?

Reply:
in the previous steps in 02-fusion-filtering.R we are using the grouping to get gene [1A/1B/2A/3B] expression per Sample so the group_by(FusionName,Sample), the Fusion_type will not be a useful group to groupby there. 

But in the aggregation step here we wanted unique [inframe/frameshift] fusion calls per Sample called by 1 or N callers so I'm using the Fusion_Type here as well so that if a fusion has 1 breakpoint that is inframe and another that is frameshift these will get a unique row to itself.

Reply:
Okay, sounds good 👍